### PR TITLE
#4732: Clean up compute_kernel_api

### PR DESCRIPTION
**Pull request by rtawfik01**

### CODE REVIEW

File: tt_metal/include/compute_kernel_api/matmul.h [hunk 1]
```diff
@@ -210,53 +189,49 @@ ALWI void matmul_block(uint32_t c_in0, uint32_t c_in1, uint32_t itile0, uint32_t
...
-    UNPACK(( llk_unpack_reconfig_data_format_srca(cbid, in1_cb_id) ));
+ALWI void mm_block_init_short(uint32_t in0_cb_id = 0, uint32_t in1_cb_id = 1, uint32_t transpose=0, uint32_t ct_dim = 1, uint32_t rt_dim = 1, uint32_t kt_dim = 1) {
+    UNPACK(( llk_unpack_AB_matmul_init(in0_cb_id, in1_cb_id, transpose, ct_dim, rt_dim, kt_dim) ));
 
     #ifdef ARCH_GRAYSKULL
-    MATH(( llk_math_matmul_init<MATH_FIDELITY, DstTileFaceLayout::ColMajor>(in0_cb_id, in1_cb_id) ));
+    MATH(( llk_math_matmul_init<MATH_FIDELITY, DstTileFaceLayout::ColMajor>(in0_cb_id, in1_cb_id, transpose, ct_dim, rt_dim, kt_dim)  ));
     #else
-    MATH(( llk_math_matmul_init<MATH_FIDELITY, DstTileFaceLayout::RowMajor>(in0_cb_id, in1_cb_id, 0, ct_dim, rt_dim, kt_dim) ));
+    MATH(( llk_math_matmul_init<MATH_FIDELITY, DstTileFaceLayout::RowMajor>(in0_cb_id, in1_cb_id, transpose, ct_dim, rt_dim, kt_dim)  ));
```
Comment (mywoodstock) [inline tt_metal/include/compute_kernel_api/matmul.h:211]: these look the same, can remove ifdef?

Comment (mywoodstock) [reply tt_metal/include/compute_kernel_api/matmul.h:211]: oh nvm, the facelayout is different :)

**Review by mywoodstock — APPROVED** [review]
this is awesome, thanks!!

File: tt_metal/include/compute_kernel_api/matmul.h (hunk 1 above)
Comment (rtawfik01) [reply tt_metal/include/compute_kernel_api/matmul.h:211]: yeah one is colmajor and the other is rowmajor, my next commit will try to hide that, but in the meantime I am doing incremental clean ups so I don't break pipelines

**Comment by rtawfik01** [comment]
Will push once this pipeline passes: https://github.com/tenstorrent-metal/tt-metal/actions/runs/7906811044/job/21582491611

**Comment by rtawfik01** [comment]
Passed unit tests here: https://github.com/tenstorrent-metal/tt-metal/actions/runs/7907284670/job/21583999157
perf tests failed due to test_tracer.py bug on main, but passes locally

**Pull request merged**